data/jobfactory.py

```diff
@@ -20,16 +22,41 @@ def create_job_factory(job_answer_set):
     job_name = job_answer_set.job_name
     vm_project_name = job_answer_set.questionnaire.vm_project.name
     vm_flavor_name = job_answer_set.questionnaire.vm_flavor.name
-    volume_size = job_answer_set.questionnaire.volume_size
+    volume_size = calculate_volume_size(job_answer_set)
     share_group = job_answer_set.questionnaire.share_group
     fund_code = job_answer_set.fund_code
-
     factory = JobFactory(user, workflow_version, stage_group, user_job_order_dict, system_job_order_dict, job_name, vm_project_name,
                          vm_flavor_name, volume_size, share_group, fund_code)
 
     return factory
 
 
+def calculate_volume_size(job_answer_set):
+    """
+    Calculates the volume size needed based on the job_answer_set questionnaire settings and stage group data.
+    :param job_answer_set: JobAnswerSet: contains questionnaire and stage_group used in calculation
+    :return: int: size in GB: volume_size_factor * data_size_in_gb + volume_size_base
+    """
+    base_in_gb = job_answer_set.questionnaire.volume_size_base
+    factor = job_answer_set.questionnaire.volume_size_factor
+    data_size_in_gb = calculate_stage_group_size(job_answer_set.stage_group)
+    return int(round(base_in_gb + float(factor) * data_size_in_gb))
```

Comment:
Wouldn't want to `ceil` instead of round? I hope 1GB wouldn't break things either way, but it seems like if we calculate 3.4GB that 3.0GB is too little.